src/base/orderedset.h

```diff
@@ -100,9 +100,18 @@ class OrderedSet : public std::set<T, Compare>
         return (BaseType::erase(value) > 0);
     }
 
-    ThisType &unite(const ThisType &other)
+    template <typename Iterable>
+    ThisType &unite(const Iterable &other)
     {
         BaseType::insert(other.cbegin(), other.cend());
         return *this;
     }
+
+    template <typename Iterable>
+    ThisType united(const Iterable &other) const
```

Comment:
please consider add/update tests in testorderedset.cpp.

Reply:
> please consider add/update tests in testorderedset.cpp.

Done.